src/CatalystClient.ts

```diff
@@ -1,30 +1,34 @@
 import { Entity } from '@dcl/schemas'
-import { Fetcher, HealthStatus, RequestOptions } from 'dcl-catalyst-commons'
+import { IFetchComponent } from '@well-known-components/http-server'
+import { HealthStatus } from 'dcl-catalyst-commons'
 import { CatalystAPI } from './CatalystAPI'
 import { BuildEntityOptions, BuildEntityWithoutFilesOptions, ContentClient } from './ContentClient'
 import { EmotesFilters, OwnedItems, ServerMetadata, WearablesFilters } from './LambdasAPI'
 import { LambdasClient } from './LambdasClient'
+import {
+  DeploymentBuilder,
+  DeploymentData,
+  DeploymentPreparationData,
+  RequestOptions,
+  createFetchComponent,
+  sanitizeUrl
+} from './utils'
 import { clientConnectedToCatalystIn } from './utils/CatalystClientBuilder'
-import { DeploymentBuilder, DeploymentData, DeploymentPreparationData } from './utils/DeploymentBuilder'
-import { getHeadersWithUserAgent, sanitizeUrl } from './utils/Helper'
 
 export type CatalystClientOptions = {
   catalystUrl: string
-  fetcher?: Fetcher
+  fetcher?: IFetchComponent
   deploymentBuilderClass?: typeof DeploymentBuilder
 }
+
 export class CatalystClient implements CatalystAPI {
   private readonly contentClient: ContentClient
   private readonly lambdasClient: LambdasClient
   private readonly catalystUrl: string
 
   constructor(options: CatalystClientOptions) {
     this.catalystUrl = sanitizeUrl(options.catalystUrl)
-    const fetcher = options.fetcher
-      ? options.fetcher
-      : new Fetcher({
-          headers: getHeadersWithUserAgent('catalyst-client')
-        })
+    const fetcher = options.fetcher ? options.fetcher : createFetchComponent()
```

Comment:
Is there a way to pass on the user-agent header? So it doesn't change behavior with regards to the old implementation